cmd/kube-apiserver/app/server.go

```diff
@@ -104,7 +109,7 @@ cluster's shared state through which all other components interact.`,
 
 			// Activate logging as soon as possible, after that
 			// show flags with the final logging configuration.
-			if err := s.Logs.ValidateAndApply(); err != nil {
+			if err := s.Logs.ValidateAndApply(utilfeature.DefaultFeatureGate); err != nil {
```

Comment:
this is a good example of a component that doesn't have a config file yet, registering the logging features in init()

users that wanted contextual logging would enable it with the CLI flag, which is parsed/applied before RunE is called